plumbing/format/diff/unified_encoder.go

```diff
@@ -73,102 +57,78 @@ func (e *UnifiedEncoder) SetColor(colorConfig ColorConfig) *UnifiedEncoder {
 	return e
 }
 
+// Encode encodes patch.
 func (e *UnifiedEncoder) Encode(patch Patch) error {
-	e.printMessage(patch.Message())
+	buf := &bytes.Buffer{}
```

Comment:
why it's not `strings.Builder` at the end?

Reply:
Good question!

The honest answer is that the code uses `fmt.Fprintf` a lot and I didn't realize that this could write to `strings.Builder` so I stuck with `bytes.Buffer`.

The post-hoc justification is that eventually we call `buf.WriteTo` to write the contents to the `io.Writer` associated with the `UnifiedEncoder`. `strings.Builder` has no such method, so we'd have to add a `string` to `[]byte` conversion.

I did also consider writing to the `io.Writer` associated with the `UnifiedEncoder` directly, but rejected it because:
* We use a lot of `WriteByte` and `WriteString` calls, which `io.Writer` does not implement.
* The accumulate-everything-in-a-`bytes.Buffer` before writing it has nice behavior when errors occur. Specifically, you don't risk writing a partial diff (unless `buf.WriteTo` fails, of course).

Reply:
`strings.Builder` was designed ad-hoc for this use case. And it's way more gentle with the memory than `bytes.Buffer`. In `strings.Builder` you have WriteByte, WriteRune, and WrityString, so replace it shouldn't be a problem. 

I don't see where are the WriteTo calls, just the method ` writeTo(buf *bytes.Buffer, color ColorConfig)` we we can change the args to the builder.

Reply:
Thanks for the info, I've updated the PR to use `strings.Builder`.